pkg/reconciler/v1alpha1/pipeline/resources/dag_test.go

```diff
@@ -96,28 +123,47 @@ func TestBuild(t *testing.T) {
 					"z": {Task: zDependsOnX, Prev: []*Node{nodeX}},
 				},
 			},
-			false,
-			""},
-		{"self-link",
-			v1alpha1.PipelineSpec{Tasks: []v1alpha1.PipelineTask{selfLink}},
-			nil,
-			true,
-			` "self-link" is invalid: : Internal error: cycle detected; task "a" depends on itself`},
-		{"cycle-2",
-			v1alpha1.PipelineSpec{Tasks: []v1alpha1.PipelineTask{xDependsOnA, zDependsOnX, aDependsOnZ}},
-			nil,
-			true,
-			` "cycle-2" is invalid: : Internal error: cycle detected; a -> x -> z -> a `},
-		{"duplicate-tasks",
-			v1alpha1.PipelineSpec{Tasks: []v1alpha1.PipelineTask{a, a}},
-			nil,
-			true,
-			` "duplicate-tasks" is invalid: spec.tasks.name: Duplicate value: "a"`},
-		{"invalid-task-name",
-			v1alpha1.PipelineSpec{Tasks: []v1alpha1.PipelineTask{invalidTask}},
-			nil,
-			true,
-			` "invalid-task-name" is invalid: spec.tasks.name: Not found: "none"`},
+			shdErr:      false,
+			expectedErr: "",
+		}, {
+			name:        "self-link",
+			spec:        v1alpha1.PipelineSpec{Tasks: []v1alpha1.PipelineTask{selfLink}},
+			expectedDAG: nil,
+			shdErr:      true,
+			expectedErr: ` "self-link" is invalid: : Internal error: cycle detected; task "a" depends on itself`,
+		}, {
+			name:        "cycle-2",
+			spec:        v1alpha1.PipelineSpec{Tasks: []v1alpha1.PipelineTask{xDependsOnA, zDependsOnX, aDependsOnZ}},
+			expectedDAG: nil,
+			shdErr:      true,
+			expectedErr: ` "cycle-2" is invalid: : Internal error: cycle detected; a -> x -> z -> a `,
+		}, {
+			name:        "duplicate-tasks",
+			spec:        v1alpha1.PipelineSpec{Tasks: []v1alpha1.PipelineTask{a, a}},
+			expectedDAG: nil,
+			shdErr:      true,
+			expectedErr: ` "duplicate-tasks" is invalid: spec.tasks.name: Duplicate value: "a"`,
+		}, {
+			name:        "invalid-task-name",
+			spec:        v1alpha1.PipelineSpec{Tasks: []v1alpha1.PipelineTask{invalidTask}},
+			expectedDAG: nil,
+			shdErr:      true,
+			expectedErr: ` "invalid-task-name" is invalid: spec.tasks.name: Not found: "none"`,
+		}, {
+			name: "no-cycle",
+			spec: v1alpha1.PipelineSpec{Tasks: []v1alpha1.PipelineTask{a, dDependsOnA, eDependsOnA, fDependsOnDAndE, gDependOnF}},
+			expectedDAG: &DAG{
+				Nodes: map[string]*Node{
+					"a": {Task: a},
+					"d": {Task: dDependsOnA, Prev: []*Node{{Task: a}}},
+					"e": {Task: eDependsOnA, Prev: []*Node{{Task: a}}},
+					"f": {Task: fDependsOnDAndE, Prev: []*Node{{Task: dDependsOnA, Prev: []*Node{{Task: a}}}, {Task: eDependsOnA, Prev: []*Node{{Task: a}}}}},
+					"g": {Task: gDependOnF, Prev: []*Node{{Task: fDependsOnDAndE, Prev: []*Node{{Task: dDependsOnA, Prev: []*Node{{Task: a}}}, {Task: eDependsOnA, Prev: []*Node{{Task: a}}}}}}},
```

Comment:
do you think it would be crazy to have some comments that try to depict the graphs in these tests? :innocent: